restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/telephony/DialActionOrganizationSBCTest.java

```diff
@@ -0,0 +1,1079 @@
+    	stubFor(post(urlPathMatching("/DialAction.*"))
+                .willReturn(aResponse()
+                    .withStatus(200)));
+    	/*
+    	 * test case 1 - bob@org2 created INVITE - sip:+12223334467@org3.restcomm.com -> call should NOT go to alice@org3 (bcz 12223334467@org3.restcomm.com is pure sip) - instead call should FAIL
+    	 */
+
+        SipURI uri = bobSipStackOrg2.getAddressFactory().createSipURI(null, "127.0.0.1:5080");
+        assertTrue(bobPhoneOrg2.register(uri, "bob", clientPassword, "sip:bob@127.0.0.1:5090", 3600, 3600));
+        Credential c = new Credential("org3.restcomm.com", "bob", clientPassword);
+        bobPhoneOrg2.addUpdateCredential(c);
+
+        //register as alice@org2.restcomm.com
+        uri = aliceSipStackOrg2.getAddressFactory().createSipURI(null, "127.0.0.1:5080");
+        assertTrue(alicePhoneOrg2.register(uri, "alice", "1234", "sip:alice@127.0.0.1:5091", 3600, 3600));
+        SipCall aliceCallOrg2 = alicePhoneOrg2.createSipCall();
+        aliceCallOrg2.listenForIncomingCall();
+
+        // bob@org2.restcomm.com - dials a pure sip number in org3.
+        final SipCall bobCallOrg2 = bobPhoneOrg2.createSipCall();
+        bobCallOrg2.initiateOutgoingCall(bobContactOrg2, pureSipNumberOrg3, null, body, "application", "sdp", null, null);
+        assertLastOperationSuccess(bobCallOrg2);
+        assertTrue(bobCallOrg2.waitForAuthorisation(3000));
+
+        assertTrue(bobCallOrg2.waitOutgoingCallResponse(5 * 1000));
+        final int response = bobCallOrg2.getLastReceivedResponse().getStatusCode();
+        logger.info("bobCallOrg2 response: "+response);
+        assertEquals(Response.NOT_FOUND, response);
+    }
+
+    /**
+     * testDialNumberExistingInMultipleOrganization:
+     *
+     * given we have 5 clients:
+     * 1. alice @ org2.
+     * 2. alice @ org3.
+     * 3. bob @ org2.
+     * 4. bob @ org3.
+     * 5. alice @ defaultOrg.
+     *
+     * we have 2 number:
+     * +12223334467@org2.restcomm.com is provider number and mapped on dial action to call alice@org2.
+     * +12223334467@org3.restcomm.com is pure sip number and mapped on dial action to call alice@org3.
+     *
+     * test case 1: bob@org2 created INVITE - sip:+12223334467@org3.restcomm.com -> call should NOT go to alice@org3 (bcz 12223334467@org3.restcomm.com is pure sip) - instead call should FAIL
+     * test case 2: bob@org2 created INVITE - sip:+12223334467@default.restcomm.com -> call should FAIL (bcz defaulOrg does not have that number)
+     * test case 3: bob@org3 created INVITE - sip:+12223334467@org2.restcomm.com(a conference in org2) -> able to join conference (bcz 12223334467@org2.restcomm.com is provider number)
+     * @throws ParseException
+     * @throws InterruptedException
+     * @throws UnknownHostException
+     */
+    @Test
+    public void testDialNumberExistingInMultipleOrganizationCase2() throws ParseException, InterruptedException, UnknownHostException {
```

Comment:
this maybe an "alternate" category test case

Reply:
I copied the DialActionOrganizationTest.java, not sure if it was present initially there